### DISCUSSION TITLE
Why do we need to write task plugins data to db？为什么需要把插件写入数据库中？

### DISCUSSION THREAD

**Discussion by leileih** [apache/dolphinscheduler#15210] [Q&A]
Why do we need to write task plugins to the database, and where is the data from t_ds_plugin_define used?
请问为什么需要把任务插件的数据写入数据库中，在哪里用到了 t_ds_plugin_define 中的数据？

**Comment by vincentx11** [comment]
There‘re various plugins in DS, such as alert and task. When ui rendering plugins, it will request specified plugin's config info first, say json, in  t_ds_plugin_define. Then ui render the form base on these configs. I've found these requests in Alert settings, check [this line ](https://github.com/apache/dolphinscheduler/blob/3.2.2-release/dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/PluginDefineMapper.xml#L44)
<br/>
![image](https://github.com/user-attachments/assets/a3fc00ca-f180-449c-8316-d0756d351262)


Unfortunately, I cannot get the same request when rendering my customer task plugin, in which should get the params defined in codes. Someone who knows how to auto-render customer task plugin's params, please give me a thread. Besides, I've checked the exist plugins info in t_ds_plugin_define that defined in my task plugin. Help!
